packages/good-design/src/apps/ubi/wizard/ClaimWizard.tsx

```diff
@@ -90,20 +90,21 @@ const WizardWrapper: FC<PropsWithChildren<{ skipOffer: Error | boolean | undefin
       {error ? <ErrorModal error={error} onClose={handleClose} overlay="dark" /> : null}
 
       {isClaiming && withSignModals ? (
-        <TxModal type="sign" customTitle={customTitle} isPending={isClaiming} />
+        <TxModal type="signMultiClaim" customTitle={customTitle} isPending={isClaiming} />
       ) : remainingClaims !== undefined ? (
         <TxModal
           type="send"
           isPending={
             !isClaimingDone &&
             remainingClaims > 0 &&
-            claimReceipts?.every((tx: ethers.providers.TransactionReceipt) => tx?.confirmations > 0)
+            claimReceipts?.every(
+              (tx: ethers.providers.TransactionReceipt) => isEmpty(tx?.confirmations) || tx?.confirmations > 0
```

Comment:
explain the logic here

Reply:
handling the scenario where one transaction fails, the other one is awaiting confirmation.
else it pre-maturely closed the modal

Reply:
are you sure empty(confirmations) is the correct check for a failed tx? not tx.status or something?

Reply:
also the name isPending doesnt seem to match the logic. the logic check if tx is no longer pending but confirmed

Reply:
it's the correct check when a transactions fails on rpc-level.

Reply:
This got removed. I don't remember why it was added, and during testing was not able to figure out any difference with having it added or not